src/selectedtests/app/app.py

```diff
@@ -39,4 +60,14 @@ def create_app(mongo_wrapper: MongoWrapper, evg_api: EvergreenApi) -> FastAPI:
     )
     app.state.db = mongo_wrapper
     app.state.evg_api = evg_api
+
+    @app.exception_handler(Exception)
+    async def uncaught_exception_handler(request: Request, exc: Exception) -> JSONResponse:
```

Comment:
@ancostas thanks for recommending this, as discussed offline, the `tenacity.RetryError` was not being logged when the log format was json for some reason, even though it was being logged when the log format was text. Anyway, this ensures that the RetryError is logged even when the log format is json:
```
{"message": "Starting fetching test_mappings for project", "lineno": 76, "filename": "project_test_mappings_controller.py", "project": "project", "logger": "selectedtests.app.controllers.project_test_mappings_controller", "level": "info"}
{"message": "An exception occurred during this request", "lineno": 28, "filename": "app.py", "exception": "RetryError[<Future at 0x10bfc1748 state=finished raised HTTPError>]", "stack_trace": "Traceback (most recent call last):\n  File \"/Users/lydia.stepanek/Library/Caches/pypoetry/virtualenvs/selectedtests-jPCSPjxc-py3.7/lib/python3.7/site-packages/starlette/middleware/errors.py\", line 156, in __call__\n    await self.app(scope, receive, _send)\n  File \"/Users/lydia.stepanek/Library/Caches/pypoetry/virtualenvs/selectedtests-jPCSPjxc-py3.7/lib/python3.7/site-packages/starlette/exceptions.py\", line 73, in __call__\n    raise exc from None\n  File \"/Users/lydia.stepanek/Library/Caches/pypoetry/virtualenvs/selectedtests-jPCSPjxc-py3.7/lib/python3.7/site-packages/starlette/exceptions.py\", line 62, in __call__\n    await self.app(scope, receive, sender)\n  File \"/Users/lydia.stepanek/Library/Caches/pypoetry/virtualenvs/selectedtests-jPCSPjxc-py3.7/lib/python3.7/site-packages/starlette/routing.py\", line 590, in __call__\n    await route(scope, receive, send)\n  File \"/Users/lydia.stepanek/Library/Caches/pypoetry/virtualenvs/selectedtests-jPCSPjxc-py3.7/lib/python3.7/site-packages/starlette/routing.py\", line 208, in __call__\n    await self.app(scope, receive, send)\n  File \"/Users/lydia.stepanek/Library/Caches/pypoetry/virtualenvs/selectedtests-jPCSPjxc-py3.7/lib/python3.7/site-packages/starlette/routing.py\", line 41, in app\n    response = await func(request)\n  File \"/Users/lydia.stepanek/Library/Caches/pypoetry/virtualenvs/selectedtests-jPCSPjxc-py3.7/lib/python3.7/site-packages/fastapi/routing.py\", line 129, in app\n    raw_response = await run_in_threadpool(dependant.call, **values)\n  File \"/Users/lydia.stepanek/Library/Caches/pypoetry/virtualenvs/selectedtests-jPCSPjxc-py3.7/lib/python3.7/site-packages/starlette/concurrency.py\", line 25, in run_in_threadpool\n    return await loop.run_in_executor(None, func, *args)\n  File \"/Users/lydia.stepanek/.pyenv/versions/3.7.3/lib/python3.7/concurrent/futures/thread.py\", line 57, in run\n    result = self.fn(*self.args, **self.kwargs)\n  File \"/Users/lydia.stepanek/mongo/selected-tests/src/selectedtests/app/controllers/project_test_mappings_controller.py\", line 77, in get\n    evg_project = try_retrieve_evergreen_project(project, evg_api)\n  File \"/Users/lydia.stepanek/mongo/selected-tests/src/selectedtests/app/evergreen.py\", line 17, in try_retrieve_evergreen_project\n    evergreen_project = get_evg_project(api, project)\n  File \"/Users/lydia.stepanek/mongo/selected-tests/src/selectedtests/evergreen_helper.py\", line 16, in get_evg_project\n    for evergreen_project in evg_api.all_projects():\n  File \"/Users/lydia.stepanek/Library/Caches/pypoetry/virtualenvs/selectedtests-jPCSPjxc-py3.7/lib/python3.7/site-packages/evergreen/api.py\", line 279, in all_projects\n    project_list = self._paginate(url)\n  File \"/Users/lydia.stepanek/Library/Caches/pypoetry/virtualenvs/selectedtests-jPCSPjxc-py3.7/lib/python3.7/site-packages/evergreen/api.py\", line 165, in _paginate\n    response = self._call_api(url, params)\n  File \"/Users/lydia.stepanek/Library/Caches/pypoetry/virtualenvs/selectedtests-jPCSPjxc-py3.7/lib/python3.7/site-packages/tenacity/__init__.py\", line 292, in wrapped_f\n    return self.call(f, *args, **kw)\n  File \"/Users/lydia.stepanek/Library/Caches/pypoetry/virtualenvs/selectedtests-jPCSPjxc-py3.7/lib/python3.7/site-packages/tenacity/__init__.py\", line 358, in call\n    do = self.iter(retry_state=retry_state)\n  File \"/Users/lydia.stepanek/Library/Caches/pypoetry/virtualenvs/selectedtests-jPCSPjxc-py3.7/lib/python3.7/site-packages/tenacity/__init__.py\", line 332, in iter\n    six.raise_from(retry_exc, fut.exception())\n  File \"<string>\", line 3, in raise_from\ntenacity.RetryError: RetryError[<Future at 0x10bfc1748 state=finished raised HTTPError>]\n", "logger": "selectedtests.app.app", "level": "error"}
```